File: woodstorage/src/main/java/com/jordylangen/woodstorage/view/WoodStorageContract.java
package com.jordylangen.woodstorage.view;

import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import com.jordylangen.woodstorage.LogEntry;

public interface WoodStorageContract {

    interface View extends Contract.View {

        void add(LogEntry logEntry);

        void addAt(LogEntry logEntry, int index);

        void clear();

        void showSnackbar(@StringRes int resId, Object... formatArgs);
    }

    interface Presenter extends Contract.Presenter<View> {

        void onOptionsItemSelected(int itemId);

        void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults);
    }
}

File: woodstorage/src/main/java/com/jordylangen/woodstorage/view/WoodStorageContract.kt
package com.jordylangen.woodstorage.view

import com.jordylangen.woodstorage.storage.LogEntry

interface WoodStorageContract {

    interface View : Contract.View {

        fun add(logEntry: LogEntry)

        fun addAt(logEntry: LogEntry, index: Int)

        fun clear()

        fun showTagFilterDialog()
    }

    interface Presenter : Contract.Presenter<View> {

        fun onOptionsItemSelected(itemId: Int)
    }
}
